python/graphstorm/dataloading/dataset.py

```diff
@@ -787,3 +790,74 @@ def infer_idxs(self):
         """ Set of nodes to do inference.
         """
         return self._infer_idxs
+
+class GSDistillData(Dataset):
+    """ Dataset for distillation
+
+    Parameters
+    ----------
+    file_list : list of str
+        List of input files.
+    tokenizer : transformers.AutoTokenizer
+        HuggingFace Tokenizer.
+    device : str
+        Device name.
+    """
+    def __init__(self, file_list, tokenizer, device):
+        super().__init__()
+        self.file_list = file_list
+        self.tokenizer = tokenizer
+        self.device = device
+        self.token_id_inputs, self.labels = self.get_inputs()
+
+    def get_inputs(self):
+        """ Tokenize textual data."""
+        for i, file_name in enumerate(self.file_list):
+            if i == 0:
+                inputs = pd.read_parquet(file_name)
```

Comment:
do we assume the input text data are stored in parquet files?

Reply:
This is the data after joining textual data with embeddings. User doesn't need to care about this and I make it be parquet file. The textual data user needs to provide is json format in this PR. There's a TODO comment to support parquet format as well from user.